### PR TITLE
feat: Add support for GET invite endpoint

### LINKED ISSUE #803: [REQUEST] Add http method to fetch invites
### Describe the feature.

There is currently no way to get an invite by code. I will open a PR that addresses this.
[link to the documentation for this](https://discord.com/developers/docs/resources/invite#get-invite)

### Code of Conduct

- [X] I agree to follow the contribution requirements.

### PR DESCRIPTION
**Pull request by Nanrech**
## About

This pull request adds support for the GET invite endpoint and moves the DELETE invite one to a new class, InviteRequest, from ChannelRequest.

## Checklist

- [ ] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - #803 
- [x] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/http/invite.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from typing import Optional, Union
+
+from ...api.cache import Cache
+from ..models.guild import Invite
+from .request import _Request
+from .route import Route
+
+
```
Comment (sillyElvy) [inline interactions/api/http/invite.py:9]:
```suggestion
__all__ = ["InviteRequest"] 


```

File: interactions/api/models/guild.py [hunk 2]
```diff
@@ -2107,6 +2114,11 @@ def __init__(self, **kwargs):
             if self._json.get("channel")
             else None
         )
+        self.guild_scheduled_event = ScheduledEvents(
+            **self._json.get("guild_scheduled_event")
+            if self._json.get("guild_scheduled_event")
+            else None
+        )
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]:
```suggestion
        self.guild_scheduled_event = ( ScheduledEvents(
            **self._json.get("guild_scheduled_event"))
            if isinstance(self._json.get("guild_scheduled_event") , dict)
            else self._json.get("guild_scheduled_event")
        )
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Resolve the ci failure please

**Comment by Nanrech** [comment]
pre-commit.ci autofix

**Comment by sillyElvy** [comment]
> pre-commit.ci autofix

https://results.pre-commit.ci/run/github/320551108/1654065877.uzBWBZ6wQuSZfWQ_YYNhuw

**Comment by Nanrech** [comment]
Ok, it should be good to go now.

**Review by sillyElvy — APPROVED** [review]

**Pull request merged**